linq.go

```diff
@@ -1291,3 +1291,56 @@ func (q Query) MaxFloat64() (min float64, err error) {
 	}
 	return q.values[maxIndex].(float64), nil
 }
+
+// SelectMany returns flattens the resulting sequences into one sequence.
+//
+// Example:
+// 	names, err := From(parents).SelectMany(func (p T, idx int) (T, error) {
+//		return p.(*Parent).Children, nil
+// 	}).Results()
+func (q Query) SelectMany(f func(T, int) (T, error)) (r Query) {
```

Comment:
can you move these methods to just after `Select()`? It logically makes sense to have them next to each other. (same with plinq.go)